sentry/src/access.rs

```diff
@@ -0,0 +1,147 @@
+use chrono::Utc;
+use futures::compat::Future01CompatExt;
+use futures::future::try_join_all;
+use redis::aio::SharedConnection;
+
+use primitives::event_submission::{RateLimit, Rule};
+use primitives::sentry::Event;
+use primitives::Channel;
+
+use crate::Session;
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum Response {
+    OnlyCreatorCanCloseChannel,
+    ChannelIsExpired,
+    ChannelIsPastWithdrawPeriod,
+    RulesError(String),
+    Success,
+}
+
+// @TODO: Make pub(crate)
+pub async fn check_access(
+    redis: &SharedConnection,
+    session: &Session,
+    rate_limit: &RateLimit,
+    channel: &Channel,
+    events: &[Event],
+) -> Response {
+    let is_close_event = |e: &Event| match e {
+        Event::Close => true,
+        _ => false,
+    };
+
+    // Check basic access rules
+    // only the creator can send a CLOSE
+    if session.uid != channel.creator && events.iter().any(is_close_event) {
+        return Response::OnlyCreatorCanCloseChannel;
+    }
+
+    let current_time = Utc::now();
+
+    if current_time > channel.valid_until {
+        return Response::ChannelIsExpired;
+    }
+
+    if current_time > channel.spec.withdraw_period_start && !events.iter().all(is_close_event) {
+        return Response::ChannelIsPastWithdrawPeriod;
+    }
+
```

Comment:
There is a missing check for not rate-limiting channel creator

Reply:
1 line bellow? Isn't this what you mean?
https://github.com/AdExNetwork/adex-validator/blob/05-stable/services/sentry/lib/access.js#L34

Reply:
It's missing in `05-stable` but a channel creator shouldn't be rate-limited
https://github.com/AdExNetwork/adex-validator/blob/master/services/sentry/lib/access.js#L37